edb_stat_statements/edb_stat_statements.c

```diff
@@ -608,14 +603,31 @@ edbss_extract_stmt_info(const char* query_str, int query_len) {
 				.object_field_start = edbss_json_ofield_start,
 				.scalar = edbss_json_scalar,
 		};
-		JsonParseErrorType parse_rv = pg_parse_json(lex, &sem);
-		freeJsonLexContext(lex);
-
-		if (parse_rv == JSON_SUCCESS)
-			if ((state.found & EDB_STMT_INFO_PARSE_REQUIRED) == EDB_STMT_INFO_PARSE_REQUIRED)
-				if (info->query_id != UINT64CONST(0))
-					return info;
 
+		while (info_str) {
+			JsonLexContext *lex = makeJsonLexContextCstringLen(
+#if PG_VERSION_NUM >= 170000
+					NULL,
+					info_str,
+#else
+					(char *) info_str,  // not actually mutating
+#endif
+					info_len,
+					PG_UTF8,
+					true);
+			JsonParseErrorType parse_rv = pg_parse_json(lex, &sem);
+			freeJsonLexContext(lex);
+
+			if (parse_rv == JSON_SUCCESS)
```

Comment:
Does the state get mutated even if there is an error? I guess that's probably fine?

What happens to `info_len` on an error? Is it still updated with however much got consumed?

Reply:
> Does the state get mutated even if there is an error? I guess that's probably fine?

Yes, it'll keep the parsed values and continue to the next line, which I think is fine.

> What happens to `info_len` on an error? Is it still updated with however much got consumed?

On error, `info_len` will be updated to skip the whole failing line and restart on the next line.